### PR TITLE
D3 Graph Zoom #78

### PR DESCRIPTION
**Pull request by lposo**
The graph can now be zoomed in/out and panned.

### CODE REVIEW

File: Client/src/Components/Graph/Graph.js [hunk 1]
```diff
@@ -51,51 +32,50 @@ datalist.push(dataset3)
...
 
 class Graph extends Component {
+
     componentDidMount() {
         this.graphCreation(datalist)
     }
     graphCreation(datalist) {
         //Calls the function to create the axis
-        startLinear();
+        //startLinear();
```
Comment (TiraFire) [inline Client/src/Components/Graph/Graph.js]: You can just remove the function if it is not used. A new function will be added later for the ln to log etc.

File: Client/src/Components/Graph/Graph.js [hunk 2]
```diff
@@ -128,7 +108,40 @@ class Graph extends Component {
             .attr("cy", function (d) {
                 return yScale(d["y"]);
             })
+            .attr("clip-path", "")
```
Comment (TiraFire) [inline Client/src/Components/Graph/Graph.js]: You can remove this part too, i dont think it needs to be here the clip path attr

File: Client/src/Components/Graph/Graph.js [hunk 3]
```diff
@@ -128,7 +108,40 @@ class Graph extends Component {
...
+            var newX = d3.event.transform.rescaleX(xScale);
+            var newY = d3.event.transform.rescaleY(yScale);
+
+            xAxis.call(d3.axisBottom(newX))
+            yAxis.call(d3.axisLeft(newY))
+
+            scatter
+                .selectAll("circle")
+                .attr('cx', function (d) { return newX(d["x"]) })
+                .attr('cy', function (d) { return newY(d.["y"]) });
```
Comment (TiraFire) [inline Client/src/Components/Graph/Graph.js]: please remove the "." in the d.["y"]

**Review by TiraFire — CHANGES_REQUESTED** [review]
My small refactoring should be merged soon, so youll have to merge dev into yours before merging into dev. It might need some rework since it changes some functions.

Overall, great work!

**Comment by lposo** [comment]
Added the requested changes. I'll wait until the other PR is finished.

**Review by TiraFire — APPROVED** [review]

File: Client/src/Components/Graph/Graph.js [hunk 4]
```diff
@@ -51,51 +32,49 @@ datalist.push(dataset3)
 datalist.push(dataset4)
 
 class Graph extends Component {
```
Comment (GD-Dheer) [inline Client/src/Components/Graph/Graph.js]: Mount your changes after pulling Tiras latest changes

**Review by GD-Dheer — COMMENTED** [review]
Merge conflicts needs to be addressed first and then the view can be tested locally. Other than that it looks good

File: Client/src/Components/Graph/Graph.js (hunk 4 above)
Comment (lposo) [reply Client/src/Components/Graph/Graph.js]: Develop branch is now merged into the zoom branch.

**Review by GD-Dheer — APPROVED** [review]
Works great! Can we remove the why am I here?
Also maybe a border or something to this graph?

**Pull request merged**